server/src/controllers/room.js

```diff
@@ -0,0 +1,33 @@
+const Boom = require('@hapi/boom');
+
+const { verifyCookie } = require('../utils/jwt');
+const { roomSchema } = require('./validation/roomSchema');
+const { addNewRoom, getRoom } = require('../database/queries');
+
+const addRoom = (req, res, next) => {
+  const { name } = req.body;
+  roomSchema
+    .validateAsync({ name })
+    .catch((error) => {
+      throw Boom.badRequest(error.message);
+    })
+    .then(() => verifyCookie(req.cookies.token))
+    .then(({ role }) => {
+      if (role) {
+        return getRoom(name);
+      }
+      throw Boom.unauthorized('This action just for admin');
+    })
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        addNewRoom(name);
```

Comment:
I think you need to return the promise

Reply:
nothing return from INSERT query so the next promise would be for undefined and that's cause an error

Reply:
if you dont return a promise, the promise resolves immediately to undefined, as a result the response would be 201, and the data did not actually change